enki/robots/thymio2/Thymio2.h

```diff
@@ -68,12 +68,95 @@ namespace Enki
 		GroundSensor groundSensor0;
 		//! The ground sensor 1 (right)
 		GroundSensor groundSensor1;
+
+		static unsigned int textureDimension;
+		unsigned int textureID;
+		unsigned int* ledTexture;
+		bool ledTextureNeedUpdate;
+
+		enum LED_INDEX
+		{
+			TOP_LEFT = 0,
+			TOP_RIGHT,
+			BOTTOM_LEFT,
+			BOTTOM_RIGHT,
+
+			BUTTON_UP,
+			BUTTON_DOWN,
+			BUTTON_LEFT,
+			BUTTON_RIGHT,
+
+			RING_0,
+			RING_1,
+			RING_2,
+			RING_3,
+			RING_4,
+			RING_5,
+			RING_6,
+			RING_7,
+
+			IR_FRONT_0,
+			IR_FRONT_1,
+			IR_FRONT_2,
+			IR_FRONT_3,
+			IR_FRONT_4,
+			IR_FRONT_5,
+
+			IR_BACK_0,
+			IR_BACK_1,
+
+			LEFT_BLUE,
+			LEFT_RED,
+			RIGHT_BLUE,
+			RIGHT_RED,
+
+			LED_COUNT
+		};
 		
 	public:
 		//! Create a Thymio II
 		Thymio2();
 		//! Destructor
 		~Thymio2();
+
+		bool updateLedTexture(uint32_t* base = 0, uint32_t* diffusionMap = 0);
+		void setLedIntensity(LED_INDEX ledIndex, float intensity = 1.f);
+		void setLedIntensity(LED_INDEX ledIndex, unsigned char intensity = 255);
+		void setLedColor(LED_INDEX ledIndex, Color color = Color(1.,1.,1.,1.));
+		void setLedColor(LED_INDEX ledIndex, unsigned char r = 255, unsigned char g = 255, unsigned char b = 255, unsigned char a = 255);
+		uint32_t getColorInt(LED_INDEX ledIndex);
+		Color getColor(LED_INDEX ledIndex);
+
+
+	protected:
+		class vec2f{
```

Comment:
These two classes (vec2f and vec2i) should be templatized, with:
template<typename T> vec2 { ... };
typedef vec2<int> vec2i;
typedef vec2<float> vec2f;